python/pyiceberg/avro/writer.py

```diff
@@ -136,7 +137,7 @@ class DecimalWriter(Writer):
     scale: int = dataclassfield()
 
     def write(self, encoder: BinaryEncoder, val: Any) -> None:
-        return encoder.write_decimal_bytes(val)
+        return encoder.write(decimal_to_bytes(val, byte_length=decimal_required_bytes(self.precision)))
```

Comment:
Like the comment above, we don't need to calculate the size every time since this is in a tight loop.